packages/SidePanel/src/components/Dialog/Dialog.styles.js

```diff
@@ -59,7 +37,35 @@ export const Dialog = styled.div`
 
   ${props => {
     const width = Number.isNaN(Number(props.width)) ? props.width : `${props.width}px`;
-    const animation = props.isOpen ? slideIn(props.isSlideFromLeft) : slideOut(props.isSlideFromLeft);
+    const slideOutTransform = props.isSlideFromLeft ? "translateX(-100%)" : `translateX(100%)`;
+    const slideInTransform = "translateX(0)";
+    function slideIn() {
```

Comment:
can we extract this function outside will prevent of creating this function each time the stylecomponent function runs, you can do something like:

```js
function slideIn({ slideInTransform , slideOutTransform  l}) {
  return ...
}
```

also `props.isAnimating` should be desctructure at the top of the stylecomponent function as the style guide suggests